packages/dx-scheduler-core/src/plugins/editing-state/helpers.ts

```diff
@@ -0,0 +1,239 @@
+import moment from 'moment';
+import { RRule, rrulestr, RRuleSet, Options } from 'rrule';
+import { PureComputed } from '@devexpress/dx-core';
+import {
+  AppointmentModel, PreCommitChanges, ChangeSet, Changes, MakeDateSequence,
+} from '../../types';
+import { RECURRENCE } from '../../constants';
+
+export const deleteCurrent: PureComputed<
+  [Partial<AppointmentModel>], ChangeSet
+> = (appointmentData) => {
+  const currentSequence: Date[] = makeDateSequence(appointmentData.rRule, appointmentData.exDate, {
+    dtstart: moment.utc(appointmentData.parentData.startDate).toDate(),
+  });
+
+  if (currentSequence.length === 1) {
+    return deleteAll(appointmentData);
+  }
+
+  const currentExDate = `${moment.utc(appointmentData.startDate as Date)
+    .format('YYYYMMDDTHHmmss')}Z`;
+  const exDate = appointmentData.exDate
+    ? `${appointmentData.exDate},${currentExDate}`
+    : currentExDate;
+
+  return { changed: { [appointmentData.id!]: { exDate } } };
+};
+
+export const deleteAll: PureComputed<
+  [Partial<AppointmentModel>], ChangeSet
+> = (appointmentData) => {
+  return { deleted: appointmentData.id };
+};
+
+export const deletedCurrentAndFollowing: PureComputed<
+  [Partial<AppointmentModel>], ChangeSet
+> = (appointmentData) => {
+  const { rRule, startDate, parentData, exDate: prevExDate = '' } = appointmentData;
+
+  const initialSequence: Date[] = makeDateSequence(rRule, prevExDate, {
+    dtstart: moment.utc(parentData.startDate).toDate(),
+  });
+
+  if (initialSequence.length === 1) {
+    return deleteAll(appointmentData);
+  }
+
+  const currentChildIndex = initialSequence
+    .findIndex(date => moment(date).isSame(startDate as Date));
+
+  const options: Partial<Options> = {
```

Comment:
The `deleteCurrentAndFollowing` and `editCurrentAndFollowing` methods also have big part of the same code